frontend/src/lib/components/common/confirmationModal/UnsavedConfirmationModal.svelte

```diff
@@ -0,0 +1,47 @@
+<script lang="ts">
+	import ConfirmationModal from './ConfirmationModal.svelte'
+	import { beforeNavigate, goto } from '$app/navigation'
+	import { onDestroy } from 'svelte/internal'
+	import { dirtyStore } from './dirtyStore'
+
+	let navigationState: { from: URL; to: URL | null; cancel: () => void } | undefined = undefined
+	$: open = Boolean(navigationState)
+
+	beforeNavigate((newNavigationState) => {
+		if (
+			!navigationState &&
+			$dirtyStore &&
+			newNavigationState.to?.pathname !== newNavigationState.from.pathname
+		) {
+			navigationState = newNavigationState
+			newNavigationState.cancel()
+		}
+	})
+
+	onDestroy(() => {
+		$dirtyStore = false
+	})
+</script>
+
+<ConfirmationModal
```

Comment:
can we make confirmation modal such that:
Pressing enter confirms it
Pressing escape cancels it

Reply:
Done!